### ISSUE TITLE
Xcode 14.1 xcodebuild fails to build OCHamrest 

### ISSUE THREAD

**Issue by alexbartisro** [hamcrest/OCHamcrest#92]
Hello!

I was wondering if anyone else has this issue under Xcode 14.1 and .2. when installing under Carthage

```carthage update --platform iOS --use-xcframeworks --use-netrc --cache-builds --no-use-binaries --derived-data ./derivedData```

` shell task (/usr/bin/xcrun xcodebuild -project /Users/[myProject]/Carthage/Checkouts/OCHamcrest/Source/OCHamcrest.xcodeproj -scheme libochamcrest -configuration Release CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES archive -showBuildSettings -skipUnavailableActions) failed with exit code 6:
2022-12-16 09:07:12.460 xcodebuild[88170:616009]  DVTAssertions: Warning in /System/Volumes/Data/SWE/Apps/DT/BuildRoots/BuildRoot1/ActiveBuildRoot/Library/Caches/com.apple.xbs/Sources/DVTiOSFrameworks/DVTiOSFrameworks-21506/DTDeviceKitBase/DTDKRemoteDeviceData.m:382
Details:  (null) deviceType from 00008103-000610181EE3401E was NULL when -platform called.
Object:   <DTDKMobileDeviceToken: 0x115a07b00>
Method:   -platform
Thread:   <NSThread: 0x6000029e24c0>{number = 7, name = (null)}`

basically it fails to build `libochamcrest`.

I'm getting to OCHamcrest as a dependency of OCMockito where I set it to the latest 9.0.1

https://github.com/alexbartisro/OCMockito/tree/fix-xcode13

**Comment by jonreid** [comment]
No idea. I do
```
carthage build --configuration Release --no-skip-current --use-xcframeworks
```

The output is:
```
*** xcodebuild output can be found in /var/folders/tg/h4rcf38d6f7dx4tj_jhz74fh0000gp/T/carthage-xcodebuild.8XQTHa.log
A shell task (/usr/bin/xcrun xcodebuild -project /Users/jmreid/Developer/Hamcrest/OCHamcrest/Source/OCHamcrest.xcodeproj -scheme libochamcrest -configuration Release CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES archive -showBuildSettings -skipUnavailableActions) failed with exit code 6:
2023-03-07 18:21:47.522 xcodebuild[73552:2071843] [MT] DVTAssertions: ASSERTION FAILURE in /System/Volumes/Data/SWE/Apps/DT/BuildRoots/BuildRoot2/ActiveBuildRoot/Library/Caches/com.apple.xbs/Sources/IDEFrameworks/IDEFrameworks-21534/IDEFoundation/Execution/RunDestinations/IDERunDestinationCLI.m:566
Details:  Unhandled/unexpected case where no run destinations were produced.
Object:   <IDERunDestinationCLI>
Method:   +resolveRunDestinationsWithWorkspace:scheme:buildAction:schemeCommand:schemeTask:destinationSpecifications:architectures:timeout:runDestinationManager:deviceManager:fallbackPreferredSDK:fallbackPreferredArchitectures:skipUnsupportedDestinations:shouldSkipRunDestinationValidation:didDisambiguate:disambiguatedMatches:disambiguatedMatchesDescription:error:
Thread:   <_NSMainThread: 0x6000019c0180>{number = 1, name = main}
Hints:

Backtrace:
  0   -[DVTAssertionHandler handleFailureInMethod:object:fileName:lineNumber:assertionSignature:messageFormat:arguments:] (in DVTFoundation)
  1   _DVTAssertionHandler (in DVTFoundation)
  2   _DVTAssertionFailureHandler (in DVTFoundation)
  3   _sortDevicesForDisplay (in IDEFoundation)
  4   -[Xcode3CommandLineBuildTool _resolveRunDestinationsForBuildAction:] (in Xcode3Core)
  5   -[Xcode3CommandLineBuildTool _resolveInputOptionsWithTimingSection:] (in Xcode3Core)
  6   -[Xcode3CommandLineBuildTool run] (in Xcode3Core)
  7   XcodeBuildMain (in libxcodebuildLoader.dylib)
  8   start (in dyld)
```

**Comment by jonreid** [comment]
Note that xcodebuild itself has no such problem. `./makeXCFramework.sh` works fine.

**Comment by DivineDominion** [comment]
Works on my machine when checking out `main`.

    $ carthage build --no-skip-current --use-xcframeworks
    $ carthage build --configuration Release --no-skip-current --use-xcframeworks

(Had to install tvOS and watchOS devices first)

**Comment by jonreid** [comment]
Great news… thanks for checking, @DivineDominion!

GitHub is still running Xcode 14.2, so Carthage still fails in the workflow build. I'm going to leave this open as a reminder to try again when GitHub updates Xcode.

**Comment by jonreid** [comment]
Still 14.2 on `macos-13`, despite what their documentation says. Will try again later.

**Comment by jonreid** [comment]
After some time, I discovered that while Xcode 14.3 is available on `macOS-13`, it's not the default. Specifying which Xcode to use fixed things. I finally get to close this 😌
